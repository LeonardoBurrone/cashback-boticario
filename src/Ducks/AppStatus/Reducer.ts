// TODO: testar reducer
import { Reducer } from 'redux';

import { AppStatusAction, AppStatusActionTypes, AppStatusState } from './types';

export const initialState: AppStatusState = {
  loading: false
};

export const appStatusReducer: Reducer<AppStatusState, AppStatusAction> = (
  state = initialState,
  action: AppStatusAction
): AppStatusState => {
  const { payload } = action;

  switch (action.type) {
    case AppStatusActionTypes.CHANGE_LOADING:
      return {
        ...state,
        loading: payload
      };

    default:
      return state;
  }
};

export default appStatusReducer;
