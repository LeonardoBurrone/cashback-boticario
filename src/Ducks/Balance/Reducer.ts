import { Reducer } from 'redux';

import { BalanceActionTypes, BalanceState } from './types';

export const initialState: BalanceState = {
  balance: '',
  balanceError: false
};

export const balanceReducer: Reducer<BalanceState> = (state = initialState, action): BalanceState => {
  const { payload } = action;

  switch (action.type) {
    case BalanceActionTypes.CHANGE_BALANCE:
      return {
        ...state,
        balance: payload
      };

    case BalanceActionTypes.CHANGE_BALANCE_ERROR:
      return {
        ...state,
        balanceError: payload
      };

    default:
      return state;
  }
};

export default balanceReducer;
