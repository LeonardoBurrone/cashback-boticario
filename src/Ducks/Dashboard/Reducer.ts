import { Reducer } from 'redux';

import { DashboardActionTypes, DashboardState } from './types';

export const initialState: DashboardState = {
  fetchPurchasesError: false,
  purchases: []
};

export const dashboardReducer: Reducer<DashboardState> = (state = initialState, action): DashboardState => {
  const { payload } = action;

  switch (action.type) {
    case DashboardActionTypes.FETCH_PURCHASES_ERROR:
      return {
        ...state,
        fetchPurchasesError: payload
      };

    case DashboardActionTypes.CHANGE_PURCHASES:
      return {
        ...state,
        purchases: payload
      };

    default:
      return state;
  }
};

export default dashboardReducer;
