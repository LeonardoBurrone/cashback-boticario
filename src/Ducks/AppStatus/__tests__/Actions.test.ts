import configureMockStore from 'redux-mock-store';
import {
  changeIsLoggedInAction,
  changeLoadingAction,
  changeLoadingMessageAction,
  changeRequestErrorAction
} from '../Actions';
import { initialState } from '../Reducer';

describe('Test AppStatus Actions', () => {
  const mockStore = configureMockStore();
  const store = mockStore(initialState);

  beforeEach(() => {
    store.clearActions();
  });

  it('Check if "changeIsLoggedInAction" was dispatched', () => {
    store.dispatch(changeIsLoggedInAction(true));

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });

  it('Check if "changeLoadingAction" was dispatched', () => {
    store.dispatch(changeLoadingAction(true));

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });

  it('Check if "changeLoadingMessageAction" was dispatched', () => {
    store.dispatch(changeLoadingMessageAction('Loading message...'));

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });

  it('Check if "changeRequestErrorAction" was dispatched', () => {
    store.dispatch(changeRequestErrorAction(true));

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});
