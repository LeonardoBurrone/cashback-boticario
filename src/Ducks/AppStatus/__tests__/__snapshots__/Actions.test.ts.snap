// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppStatus Actions Check if "changeIsLoggedInAction" was dispatched 1`] = `
Array [
  Object {
    "payload": true,
    "type": "CHANGE_IS_LOGGED_IN",
  },
]
`;

exports[`Test AppStatus Actions Check if "changeLoadingAction" was dispatched 1`] = `
Array [
  Object {
    "payload": true,
    "type": "CHANGE_LOADING",
  },
]
`;

exports[`Test AppStatus Actions Check if "changeLoadingMessageAction" was dispatched 1`] = `
Array [
  Object {
    "payload": "Loading message...",
    "type": "CHANGE_LOADING_MESSAGE",
  },
]
`;

exports[`Test AppStatus Actions Check if "changeRequestErrorAction" was dispatched 1`] = `
Array [
  Object {
    "payload": true,
    "type": "CHANGE_REQUEST_FAILED",
  },
]
`;
